docs/guides/initialize-state-with-props.md

```diff
@@ -0,0 +1,131 @@
+---
+title: Initialize state with props
+nav: 14
+---
+
+In cases where [dependency injection](https://en.wikipedia.org/wiki/Dependency_injection) is needed, such as when a store should be initialized with props from a component, the recommended approach is to use a vanilla store with React.context.
+
+## Store creator with `createStore`
+
+```ts
+import { createStore } from 'zustand'
+
+interface BearProps {
+  bears: number
+}
+
+interface BearState extends BearProps {
+  addBear: () => void
+}
+
+type BearStore = ReturnType<typeof createBearStore>
+
+const createBearStore = (initProps?: Partial<BearProps>) => {
+  const DEFAULT_PROPS: BearProps = {
+    bears: 0,
+  }
+  return createStore<BearState>()((set) => ({
+    ...DEFAULT_PROPS,
+    ...initProps,
+    addBear: () => set((state) => ({ bears: ++state.bears })),
+  }))
+}
+```
+
+## Basic component usage
+
+```tsx
+// Provider implementation
+import { useRef } from 'react'
+
+function App() {
+  const store = useRef(createBearStore()).current
+  return (
+    <BearContext.Provider value={store}>
```

Comment:
Should we show how the context is exactly created?
```js
import { createContext } from 'react'

const BearContext = createContext()
```
(or in TS.)

Reply:
Not sure how I left that out. It definitely should be here, and it is now.